src/Build/Evaluation/Context/EvaluationContext.cs

```diff
@@ -32,17 +33,23 @@ public enum SharingPolicy
 
         internal SharingPolicy Policy { get; }
 
-        internal virtual ISdkResolverService SdkResolverService { get; } = new CachingSdkResolverService();
-        internal IFileSystem FileSystem { get; } = new CachingFileSystemWrapper(FileSystems.Default);
+        internal ISdkResolverService SdkResolverService { get; }
+        internal IFileSystem FileSystem { get; }
+        internal EngineFileUtilities EngineFileUtilities { get; }
 
         /// <summary>
         /// Key to file entry list. Example usages: cache glob expansion and intermediary directory expansions during glob expansion.
         /// </summary>
-        internal ConcurrentDictionary<string, ImmutableArray<string>> FileEntryExpansionCache = new ConcurrentDictionary<string, ImmutableArray<string>>();
+        internal ConcurrentDictionary<string, ImmutableArray<string>> FileEntryExpansionCache { get; }
```

Comment:
Why move these into the ctor? Isn't the effect the same?

Reply:
I didn't like having part of the initialization code in static blocks and another part in the constructor, so I moved everything in the constructor.